code/modules/projectiles/guns/ballistic.dm

```diff
@@ -178,12 +181,16 @@
 		playsound(src, load_empty_sound, load_sound_volume, load_sound_vary)
 	magazine.forceMove(drop_location())
 	var/obj/item/ammo_box/magazine/old_mag = magazine
+	var/tacload_result //Used to see if the tacload was successful (ie, the new magazine actually fits)
```

Comment:
```DM
	//no tacload_result var
    if(tac_load)
        if(insert_magazine(user, tac_load, FALSE))
            do_thing()
        else
            do_other_thing()
            magazine = null
    else
        magazine = null
```

saves a var and doesn't check tac_load twice.

Reply:
shouldn't insert_magazine be handling the `null` anyways though? It looks like it's ONLY null'd when `tacload && tacload_result`

Reply:
Yeah, this is better logic, I'll implement this.

Reply:
also no insert magazine shouldn't be handling the null, it's nulled whenever the magazine is removed because otherwise the gun would still have a reference to the magazine allowing it to still function even with the item "removed"